src/iso9660.rs

```diff
@@ -522,3 +519,48 @@ fn eat(buf: &mut Bytes, n: usize) -> &mut Bytes {
     buf.advance(n);
     buf
 }
+
+/// Parse path into a PathBuf with zero or more Normal components.  Convert
+/// path to relative and resolve all "." and ".." components.
+fn canonicalize_path(s: &str) -> Result<PathBuf> {
+    // empty paths are treated the same as "/" to allow round-tripping
+    let mut buf = PathBuf::with_capacity(s.len());
+    use std::path::Component::*;
+    for c in Path::new(s).components() {
+        match c {
+            Prefix(_) | RootDir | CurDir => (),
+            ParentDir => {
+                buf.pop();
+            }
+            Normal(c) => {
+                buf.push(c);
+            }
+        }
+    }
+    Ok(buf)
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_canonicalize_path() {
+        // basic
+        assert_eq!(canonicalize_path("z").unwrap().to_str().unwrap(), "z");
+        // absolute path with . and ..
+        assert_eq!(
+            canonicalize_path("/a/./../b").unwrap().to_str().unwrap(),
```

Comment:
Looking at these paths... I wonder if we really need to support them. Feels like it's OK to just require callers to provide canonical paths? Or did you have a use case in mind that would leverage this?

Reply:
I don't have a use case, but it felt awkward that `get_path()` was trying to pick-n-choose certain behaviors of `Path` and work around others.  This way we've defined and tested clear semantics.  We could restrict the semantics further, but the implementation is basically free, so I wonder whether that's worth doing.

Reply:
Ack, SGTM!